src/main/java/de/bonndan/nivio/output/map/hex/HexMap.java

```diff
@@ -63,16 +63,7 @@ public Hex hexForItem(Item item) {
      * @return a path if one could be found
      */
     public Optional<HexPath> getPath(Item start, Item target) {
-        Optional<HexPath> optional = Optional.ofNullable(pathFinder.getPath(hexForItem(start), hexForItem(target)));
-
-        optional.ifPresent(hexPath -> {
-            if (start.getGroup() != null && start.getGroup().equals(target.getGroup())) {
-                hexPath.setGroup(start.getGroup());
-            }
-            paths.add(hexPath);
-        });
-
-        return optional;
+        return Optional.ofNullable(pathFinder.getPath(hexForItem(start), hexForItem(target)));
```

Comment:
Why doesn't the Pathfinder return an Optional itself?

Reply:
Because nobody refactored it yet 😄

Reply:
I have refactored it to use Optional.